<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AttrWidget</class>
 <widget class="QWidget" name="AttrWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>494</width>
    <height>443</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="spacing">
    <number>5</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="margin">
       <number>5</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayoutAttrListTool">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="toolButtonAddAttr">
          <property name="text">
           <string>Добавить</string>
          </property>
          <property name="icon">
           <iconset resource="modeleride.qrc">
            <normaloff>:/add</normaloff>:/add</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonEditAttr">
          <property name="text">
           <string>Изменить</string>
          </property>
          <property name="icon">
           <iconset resource="modeleride.qrc">
            <normaloff>:/edit</normaloff>:/edit</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonDeleteAttr">
          <property name="text">
           <string>Удалить</string>
          </property>
          <property name="icon">
           <iconset resource="modeleride.qrc">
            <normaloff>:/remove</normaloff>:/remove</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacerAttrListTool">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxInInherited">
          <property name="text">
           <string>Унаследованные</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableView" name="tableViewAttr">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>24</number>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutPropAttr">
     <item>
      <widget class="QGroupBox" name="groupBoxPropAttr">
       <property name="title">
        <string>Свойства атрибута</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="5" column="0">
         <widget class="QLabel" name="labelIsNull">
          <property name="text">
           <string>Может быть нулевым:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="labelIsCandidateKey">
          <property name="text">
           <string>Кандидат в ключ</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelAttrDesc">
          <property name="text">
           <string>Описание:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditAttrDesc"/>
        </item>
        <item row="5" column="1">
         <widget class="QCheckBox" name="checkBoxIsNull">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEditAttrName"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelAttrName">
          <property name="text">
           <string>Имя атрибута:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelAttrGroup">
          <property name="text">
           <string>Группа:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QCheckBox" name="checkBoxIsCandidateKey">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="labelIsUnique">
          <property name="text">
           <string>Уникален:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="checkBoxIsUnique">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="comboBoxAttrGroup">
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxPropType">
       <property name="title">
        <string>Свойства типа</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBoxStringLen">
          <property name="maximum">
           <number>8000</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="labelDefaultValue">
          <property name="text">
           <string>По умолчанию:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelTypeAttr">
          <property name="text">
           <string>Тип атрибута:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelStringLen">
          <property name="text">
           <string>Длина строки:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelLinkClass">
          <property name="text">
           <string>Ссылка на класс:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="TreeComboBox" name="comboBoxLinkAttr"/>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBoxTypeAttr"/>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="lineEditDefaultValue"/>
        </item>
       </layout>
       <zorder>labelTypeAttr</zorder>
       <zorder>labelStringLen</zorder>
       <zorder>comboBoxTypeAttr</zorder>
       <zorder>labelLinkClass</zorder>
       <zorder>comboBoxLinkAttr</zorder>
       <zorder>labelDefaultValue</zorder>
       <zorder>lineEditDefaultValue</zorder>
       <zorder>spinBoxStringLen</zorder>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutAttrSave">
     <item>
      <spacer name="horizontalSpacerLeftAttr">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonAttrSave">
       <property name="text">
        <string>Сохранить</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonAttrCancel">
       <property name="text">
        <string>Отмена</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacerRightAttr">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreeComboBox</class>
   <extends>QComboBox</extends>
   <header>treecombobox/treecombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="modeleride.qrc"/>
 </resources>
 <connections/>
</ui>
